typescript/sdk/src/token/deploy.ts

```diff
@@ -135,50 +137,63 @@ export class HypERC20Deployer extends GasRouterDeployer<
     chain: ChainName,
     config: HypERC20CollateralConfig,
   ): Promise<HypERC20Collateral> {
-    let contractName:
+    let tokenType:
       | TokenType.fastCollateral
       | TokenType.collateral
       | TokenType.collateralVault;
     switch (config.type) {
       case TokenType.fastSynthetic || TokenType.fastCollateral:
-        contractName = TokenType.fastCollateral;
+        tokenType = TokenType.fastCollateral;
         break;
       case TokenType.collateral:
-        contractName = TokenType.collateral;
+        tokenType = TokenType.collateral;
         break;
       case TokenType.collateralVault:
-        contractName = TokenType.collateralVault;
+        tokenType = TokenType.collateralVault;
         break;
       default:
         throw new Error(`Unknown collateral type ${config.type}`);
     }
-    return this.deployContract(chain, contractName, [
-      config.token,
-      config.mailbox,
-    ]);
+    return this.deployContractWithName(
+      chain,
+      tokenType,
+      hypERC20contracts[tokenType],
```

Comment:
```suggestion
      hypERC20contracts[config.type],
```

Reply:
not sure what you mean by this and the other comment below

split out `tokenType` to a var because there's some logic based on `config.type` to determine the `tokenType` to deploy, and we have to make sure the `contractName` corresponds to this `tokenType`